gstreamer-sdp/src/auto/s_d_p_media.rs

```diff
@@ -0,0 +1,260 @@
+            ffi::gst_sdp_media_attributes_len(self.to_glib_none().0)
+        }
+    }
+
+    //pub fn attributes_to_caps(&self, caps: /*Ignored*/&mut gst::Caps) -> SDPResult {
+    //    unsafe { TODO: call ffi::gst_sdp_media_attributes_to_caps() }
+    //}
+
+    pub fn bandwidths_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_bandwidths_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn connections_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_connections_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn formats_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_formats_len(self.to_glib_none().0)
+        }
+    }
+
+    //pub fn get_attribute(&self, idx: u32) -> /*Ignored*/Option<SDPAttribute> {
+    //    unsafe { TODO: call ffi::gst_sdp_media_get_attribute() }
+    //}
+
+    pub fn get_attribute_val(&self, key: &str) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_attribute_val(self.to_glib_none().0, key.to_glib_none().0))
+        }
+    }
+
+    pub fn get_attribute_val_n(&self, key: &str, nth: u32) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_attribute_val_n(self.to_glib_none().0, key.to_glib_none().0, nth))
+        }
+    }
+
+    //pub fn get_bandwidth(&self, idx: u32) -> /*Ignored*/Option<SDPBandwidth> {
+    //    unsafe { TODO: call ffi::gst_sdp_media_get_bandwidth() }
+    //}
+
+    //pub fn get_caps_from_media(&self, pt: i32) -> /*Ignored*/Option<gst::Caps> {
+    //    unsafe { TODO: call ffi::gst_sdp_media_get_caps_from_media() }
+    //}
+
+    //pub fn get_connection(&self, idx: u32) -> /*Ignored*/Option<SDPConnection> {
+    //    unsafe { TODO: call ffi::gst_sdp_media_get_connection() }
+    //}
+
+    pub fn get_format(&self, idx: u32) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_format(self.to_glib_none().0, idx))
+        }
+    }
+
+    pub fn get_information(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_information(self.to_glib_none().0))
+        }
+    }
+
+    //pub fn get_key(&self) -> /*Ignored*/Option<SDPKey> {
+    //    unsafe { TODO: call ffi::gst_sdp_media_get_key() }
+    //}
+
+    pub fn get_media(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_media(self.to_glib_none().0))
+        }
+    }
+
+    pub fn get_num_ports(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_get_num_ports(self.to_glib_none().0)
+        }
+    }
+
+    pub fn get_port(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_get_port(self.to_glib_none().0)
+        }
+    }
+
+    pub fn get_proto(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_proto(self.to_glib_none().0))
+        }
+    }
+
+    pub fn init(&mut self) -> SDPResult {
```

Comment:
This makes not much sense, we rather need a proper constructor

Reply:
same